samples/camera/js/ui.js

```diff
@@ -0,0 +1,188 @@
+      } else {
+        hideProSettings(controls.lastProIcon);
+        showProSettings(id);
+        controls.lastProIcon = id;
+      }
+      findSettingsBottomValue();
+    });
+  });
+
+  // Show or hide settings by clicking on video.
+  video.addEventListener('click', function () {
+    if (controls.settingsVisible) hideProSettings(controls.lastProIcon);
+    else showProSettings(controls.lastProIcon);
+    findSettingsBottomValue();
+  });
+
+  // Event for reset button.
+  let resetButton = document.querySelector('.reset-button');
+  resetButton.addEventListener('click', function () {
+    let constraint = { advanced: [{}] };
+    if (controls.lastProIcon == 'standard') {
+      standartSettingsIds.forEach(function (id, i) {
+        controls[id].value =
+          (controls[id].max - controls[id].min) / 2 + Number(controls[id].min);
+        if (!controls[id].disabled)
+          constraint.advanced[0][id] = controls[id].value;
+      });
+    } else if (controls.lastProIcon == 'exposureTime')
+      constraint.advanced[0]['exposureMode'] = 'continuous';
+    else if (controls.lastProIcon == 'focusDistance')
+      constraint.advanced[0]['focusMode'] = 'continuous';
+    else if (controls.lastProIcon == 'colorTemperature') {
+      constraint.advanced[0]['whiteBalanceMode'] = 'continuous';
+      document.getElementById('wbIcon').innerText = 'wb_auto';
+    }
+
+    videoTrack.applyConstraints(constraint).catch(e => console.log(e));
+  });
+
+  // TakePhoto event by clicking takePhotoButton.
+  let takePhotoButton = document.getElementById('takePhotoButton');
+  takePhotoButton.addEventListener('click', function () {
+    takePhoto();
+    const canvas = document.getElementById('gallery');
+    canvas.getContext('2d').rotate(-90 * Math.PI / 180); // rotate 90 degrees
+  });
+
+  let facingModeButton = document.getElementById('facingModeButton');
+  // Switch to face or environment mode by clicking facingModeButton.
+  facingModeButton.addEventListener('click', function () {
+    if (controls.facingMode == 'user') {
+      controls.facingMode = 'environment';
+      videoConstraint.deviceId = { exact: controls.backCamera.deviceId };
+      facingModeButton.innerText = 'camera_front';
+    } else if (controls.facingMode == 'environment') {
+      controls.facingMode = 'user';
+      videoConstraint.deviceId = { exact: controls.frontCamera.deviceId };
+      facingModeButton.innerText = 'camera_rear';
+    }
+    utils.stopCamera();
+    utils.startCamera(videoConstraint, 'videoInput', startCameraProcessing);
+  });
+}
+
+function startCameraProcessing() {
+  videoTrack = video.srcObject.getVideoTracks()[0];
+  imageCapturer = new ImageCapture(videoTrack);
+  // Timeout needed in Chrome, see https://crbug.com/711524.
+  setTimeout(() => {
+    applyInitialSettings();
+  }, 500);
+}
+
+function displayColorTemperatureIcon() {
+  let id = 'colorTemperature';
+  if (controls[id].value >= 2500 && controls[id].value <= 3500)
```

Comment:
use a switch case here ?

```js

let text =""
switch(controls[id].value)
case (> 9000): 
    break
case (> 8000):
   text = 'brightness_3'; break;
..
document.getElementById('wbIcon').innerText ='wb_' + text
```
I just checked [this](http://jsfiddle.net/some/HKdug/), where its asking to use if-then or switch with immediate values and not range

We have work with ranges others wise map could have been a good choice
```js
let colorTemperature = {
  9000 : '',
  8000 : "twilight",
  6500 : "cloudy",
  ...
};
```

document.getElementById('wbIcon').innerText = Object.keys(controls[id].value)